Some wording updates to flaky-test-management.mdx (#6677)
cypress-io/cypress-documentation: docs/cloud/features/flaky-test-management.mdx

## Why Flaky Test Management

A **flaky test** passes and fails across retries without any code change. 

> In a Cypress Cloud run, the "flaky" badge is added to a test fails on the first attempt, 

Reviewer: ```suggestion
In a Cypress Cloud run, the "flaky" badge is added to a test when the test fails on the first attempt, 
```

  vague complaint. There's no way to know whether reliability is improving or
  degrading over time, or which tests are actually responsible.

### The four main sources of flake

Any test can show flake due to do a range of causes, including legitmate bugs.

- **Real bugs**: In a stable test environment with well-written tests, flake
in a certain test often represents **hidden race conditions** in the application you are
testing. These might not show up on developer machines but become material on

> lower-powered CI machines, as well as real user devices. This is the signal to watch for.

Reviewer: Just an odd ending sentence

```suggestion
lower-powered CI machines, as well as real user devices.
```

- **Test design**: Some tests are written to move quickly without verifying the
ready-state before making an assertion. This can lead to inconsistent results because
of assumptions the tests rely on about the order of events - for example, which part 
of the page loads first.
- **Unstable test environment**: End-to-end tests often execute against live
production environments or staging/QA servers that integrate many systems, which
can indpendently have downtime.
- **Third-party dependencies**: If your test requires passing through a service
that's outside of your control (such as a payment provider), it may appear flaky
due to issues in the third party.

Understanding flake helps you find and control what might be unstable, improving
your pipeline success rate, and boosting confidence in the results. It also helps you
avoid real bugs that can have financial impact to your organization. A persistent,

> unexplained flaky test should be addressed, even if it not something you want to block

Reviewer: ```suggestion
unexplained flaky test should be addressed, even if it is not something you want to block
```

- A filterable **log of every flaky test**, ordered by severity, ready to triage
  and assign.

**Cross-run flake analysis is the key** to discovering and tracking flake in a way that

> accounts indvidual run variation in results, to give you the true success rate of your s
> uite over time. Sometimes a flaky test will pass on the first

Reviewer: ```suggestion
accounts indvidual run variation in results, to give you the true success rate of your suite over time. Sometimes a flaky test will pass on the first
```

> attempt, or fail all attempts, and not have been considered flaky on that specifc run.

Reviewer: I get what you're saying here - but how would I actually account for this and act on it if it's not tagged as 'flaky'?

Author: Thanks, will look at how to frame this better. On the analytics page, I would expect even if a persistently flaky test sometimes fully passes and sometimes fully fails, it will still actually flake sometimes too, and so you'd catch it by reviewing the full list of tests that have flaked in a given time period.

  alt="Flake rate vs. failure rate over time in Cypress Cloud flaky test analytics"
/>

> A test being flagged as flaky means that, within the runs you have filtered to,

Reviewer: I don't understand why the filter is relevant here

```suggestion
A test being flagged as flaky means that, 
```

that test failed on the first attempt then passed after retries. The
status of an individual retry attempt is separate from the final test status.

For example, a project configured to retry failing tests up to three times may
fail the first two attempts and pass on the third, a final status of passing.

Because of this, a test can show a **zero final failure rate while still
exhibiting flake**. On the other hand, a test that fails all attempts is
considered failed, not flaky - it was consistent. And a test that passes first time
is considered passing, it will not be repeated.

This is exactly why flake is so easy to
miss without dedicated tracking: the build goes green, but the underlying
instability, and its cost, is still there.

> 
> 

Reviewer: ```suggestion
```
